drivers/docker/fingerprint.go

```diff
@@ -40,7 +55,10 @@ func (d *Driver) buildFingerprint() *drivers.Fingerprint {
 	}
 	client, _, err := d.dockerClients()
 	if err != nil {
-		d.logger.Info("failed to initialize client", "error", err)
+		if d.fingerprintSuccess == nil || *d.fingerprintSuccess {
```

Comment:
The write is coordinated by a mutex - but the read isn't.

Also, given that `fingerprintSuccess` field isn't used anywhere except for `handleFingerprint` function, should it be a function local variable - and then you don't need a field or a mutex for it.  I don't think we need to handle a case where we have multiple fingerprint goroutines running.